xonsh/events.py

```diff
@@ -248,22 +253,35 @@ def transmogrify(self, name, klass):
             raise ValueError("Invalid event class; must be a subclass of AbstractEvent")
 
         oldevent = getattr(self, name)
-        newevent = type(name, (klass,), {'__doc__': type(oldevent).__doc__})()
+        newevent = self._mkevent(name, klass, type(oldevent).__doc__)
         setattr(self, name, newevent)
 
         for handler in oldevent:
             newevent.add(handler)
 
     def __getattr__(self, name):
         """Get an event, if it doesn't already exist."""
+        if name.startswith('_'):
+            raise AttributeError
         # This is only called if the attribute doesn't exist, so create the Event...
-        # (A little bit of magic to enable docstrings to work right)
-        e = type(name, (Event,), {'__doc__': None})()
+        e = self._mkevent(name)
         # ... and save it.
         setattr(self, name, e)
         # Now it exists, and we won't be called again.
         return e
 
+    def clean_between_tests(self):
```

Comment:
I could name it `reset_all_events()` instead and it would do the same thing and have the same usefulness.